core/src/main/scala/kafka/Kafka.scala

```diff
@@ -64,11 +64,7 @@ object Kafka extends Logging {
   private def buildServer(props: Properties): Server = {
     val config = KafkaConfig.fromProps(props, doLog = false)
     if (config.requiresZookeeper) {
-      new KafkaServer(
-        config,
-        Time.SYSTEM,
-        threadNamePrefix = None
-      )
+      throw new RuntimeException("ZooKeeper is not supported")
```

Comment:
Can we not simply remove this class? It's not a public class and I don't think anything references it besides the relevant shell script.

Reply:
If the `KafkaServer` is still existent, we can't remove many unused classes/configs from code base. For example, zk configs and `ZkMetadataCache`. Do we have any use cases for `KafkaServer` in 4.0?

Reply:
I actually meant something else - I was asking whether we need to keep `kafka.Kafka`. The answer is `yes` since it is still used for KRaft. But looking at this in more detail, this exception message doesn't make sense in a world where ZK is not supported since it will throw this when process roles is empty.

We should simply remove this conditional code and leave it to `KafkaConfig` to throw the appropriate exception if `processRoles` is empty.

Reply:
Sorry for misunderstanding your comment 

> We should simply remove this conditional code and leave it to KafkaConfig to throw the appropriate exception if processRoles is empty.

make sense